VideoLocker/src/main/java/org/edx/mobile/user/UserAPI.java

```diff
@@ -117,7 +121,7 @@ String getUserEnrolledCoursesURL(@NonNull String username) {
             try {
                 json = IOUtils.toString(input.in(), Charset.defaultCharset());
             } catch (IOException e) {
-                throw new RetroHttpException(e);
+                throw new RetroHttpException(RetrofitError.networkError(null, e));
```

Comment:
Instead of passing `null` we should pass a valid url by calling `response.getUrl()`.